ddtrace/llmobs/_experiment.py

```diff
@@ -212,5 +249,54 @@ def _run_task(
         self._llmobs_instance.flush()  # Ensure spans get submitted in serverless environments
         return task_results
 
-    def _run_evaluators(self, task_results, raise_errors: bool = False) -> None:
-        pass
+    def _run_evaluators(self, task_results: List[_TaskResult], raise_errors: bool = False) -> List[_EvaluationResult]:
+        if not task_results:
+            raise ValueError("No task results to evaluate.")
+        evaluations: List[_EvaluationResult] = []
+        for idx, task_result in enumerate(task_results):
+            output_data = task_result["output"]
+            record: DatasetRecord = self._dataset[idx]
+            input_data = record["input_data"]
+            expected_output = record["expected_output"]
+            evals_dict = {}
+            for evaluator in self._evaluators:
+                eval_result: JSONType = None
+                eval_err: JSONType = None
+                try:
+                    eval_result = evaluator(input_data, output_data, expected_output)
+                except Exception as e:
+                    exc_type, exc_value, exc_tb = sys.exc_info()
+                    exc_type_name = type(e).__name__ if exc_type is not None else "Unknown Exception"
+                    exc_stack = "".join(traceback.format_exception(exc_type, exc_value, exc_tb))
+                    eval_err = {"message": str(exc_value), "type": exc_type_name, "stack": exc_stack}
+                    if raise_errors:
+                        raise RuntimeError(f"Evaluator {evaluator.__name__} failed on row {idx}")
+                evals_dict[evaluator.__name__] = {"value": eval_result, "error": eval_err}
+            evaluation: _EvaluationResult = {"idx": idx, "evaluations": evals_dict}
+            evaluations.append(evaluation)
+        return evaluations
+
+    def _merge_results(
+        self, task_results: List[_TaskResult], evaluations: List[_EvaluationResult]
+    ) -> List[ExperimentResult]:
+        experiment_results = []
+        for idx, task_result in enumerate(task_results):
+            output_data = task_result["output"]
+            metadata: Dict[str, JSONType] = {"tags": cast(List[JSONType], self._tags)}
+            _metadata = task_result.get("metadata") or {}
+            if isinstance(_metadata, dict):
```

Comment:
```suggestion
```
i think we can assume it is here

Reply:
Yeah this was a weird way to handle type hints, going to fix this up